documentation/expanding.md

```diff
@@ -0,0 +1,196 @@
+* _\<new-compiler-fqdn\>_ - The FQDN and certname of the new Compiler to initialize
+* _\<new-compiler-target-group\>_ - The target availability group letter to assign to the new Compiler
+* _\<target-group-server-fqdn\>_ - The FQDN and certname of the Primary Puppet server that is assigned to the new Compiler's target availability group letter
+* _\<target-group-postgresql-fqdn\>_ - The FQDN and certname of the PE-PostgreSQL server that is assigned to the new Compiler's target availability group letter
+
+## Adding External Databases with peadm::add_database
+
+An external PE-PostgreSQL database is the component which separates Extra Large from Large deployments. These external database servers are PuppetDB specific and do not serve any other data storage services to other components of Puppet Enterprise. When the Extra Large architecture is being utilized and disaster recovery (DR) is enabled, two external PE-PostgreSQL servers must be provisioned and it is required that both PE-PostgreSQL servers exist prior to provisioning a Replica because it is immediately dependent upon this second external database server. One should notice that in both variations of this action, with **no DR** plans and when preparing to enable it for the first time, the command is exactly the same. This specific plan is intelligent enough to determine the most appropriate values for your second external database by examining the values which were used for the first.
+
+### Add an external PE-PostgreSQL server in all scenarios
+
+    bolt plan run peadm::add_database -t <new-postgres-server-fqdn> primary_host=<primary-server-fqdn>
+
+## Enable Disaster Recovery and Add a Replica with peadm::add_replica
+
+All three architectures have two variations, disaster recovery (DR) enabled or disabled. This does not have an effect on the deployed architecture naming, a Standard deployment which deploys a DR Replica is still referred to as the Standard architecture. The basic process remains the same when adding a Replica to any architecture but there are differences in CLI arguments when operating on an Extra Large deployment. When DR is enabled using PEADM, a second availability group will be created, **B**. When DR is not enabled, only one exists, **A**. These availability groups are created for pairing subsets of Compilers with the backend source which will configure them, among other things. By default the initial Primary is assigned availability group **A** and the initial Replica is assigned **B**. The availability group assignments stay with the server and importantly not with the role of the server. If you choose to promote the Replica assigned to **B** to a Primary than availability group **B** is now simply the group containing the Primary. When adding a Replica you do not need to know the availability group of the Primary it is being paired with, the `peadm::add_replica` plan will determine the appropriate group by looking up which group the Primary is assigned.
```

Comment:
- Maybe we should decide on `deployment architecture` rather than `architecture` or `deployment`
- Start a new paragraph with the words `When DR is enabled`
- `Availability group` is not immediately clear what is meant here to a new user
- Same for `backend source`
- `which group the Primary is assigned` -> `to which group the Primary is assigned`

Reply:
All these are included in new commit except for bullet one. I kinda of did something different and refer to the architecture which is adopted by the current deployment of PE as "deployment architecture" but continued using "deployment" when I am referring directly to the artifact which is the collection of deployed running PE services.